components/scream/src/share/tests/common_physics_functions_tests.cpp

```diff
@@ -0,0 +1,298 @@
+    // calculate_dse(T=0.0, z=0.0) = surf_geopotential
+    // calculate_dse(T=1/cp, z=1/gravity) = surf_potential+2
+    T0=0.0; ztest=0.0; z0=10.0;
+    REQUIRE(physicscommon_host::calculate_dse(T0,ztest,z0)==10.0);
+    T0=inv_cp; ztest=1.0/ggr; z0=0.0;
+    REQUIRE(physicscommon_host::calculate_dse(T0,ztest,z0)==z0+2.0);
+    // DZ tests
+    // calculate_dz(pseudo_density=0) = 0
+    // calculate_dz(T=0) = 0
+    // calculate_dz(pseudo_density=p0,p_mid=p0,T=1.0,qv=0) = Rd/ggr
+    // calculate_dz(pseudo_density=ggr,p_mid=Rd,T=T0,qv=0) = T0
+    dp0=0.0; ptest=p0; T0=100.0; qv0=1e-5;
+    REQUIRE(physicscommon_host::calculate_dz(dp0,ptest,T0,qv0)==0.0);
+    dp0=100.0; ptest=p0; T0=0.0; qv0=1e-5;
+    REQUIRE(physicscommon_host::calculate_dz(dp0,ptest,T0,qv0)==0.0);
+    dp0=p0; ptest=p0; T0=1.0; qv0=0.0;
+    REQUIRE(physicscommon_host::calculate_dz(dp0,ptest,T0,qv0)==Rd/ggr);
+    dp0=ggr; ptest=Rd; T0=100.0; qv0=0.0;
+    REQUIRE(physicscommon_host::calculate_dz(dp0,ptest,T0,qv0)==T0);
+    
+    // Run tests on full views
+    TeamPolicy policy(ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, 1));
+    Kokkos::parallel_for("test_universal_physics", policy, KOKKOS_LAMBDA(const MemberType& team) {
+      // Constants used in testing (necessary to ensure SP build works)
+      // Exner property tests:
+      // exner_function(pressure) should work for Scalar and for Spack.
+      physicscommon::exner_function(team,pressure,exner);
+      physicscommon::exner_function(team,pressure_packed,exner_packed);
+      // Potential temperature property tests
+      // calculate_theta_from_T(T,pressure) should work
+      // calculate_T_from_theta(theta,pressure) should work for Scalar and for Spack
+      physicscommon::calculate_theta_from_T(team,temperature,pressure,theta);
+      physicscommon::calculate_T_from_theta(team,theta,pressure,T_mid_from_pot);
+      physicscommon::calculate_theta_from_T(team,temperature_packed,pressure_packed,theta_packed);
+      physicscommon::calculate_T_from_theta(team,theta_packed,pressure_packed,T_mid_from_pot_packed);
+      // Virtual temperature property tests
+      // calculate_virtual_temperature(temperature,qv) should work for Scalar and for Spack
+      // calculate_temperature_from_virtual_temperature should work for Scalar and for Spack
+      physicscommon::calculate_virtual_temperature(team,temperature,qv,T_virtual);
+      physicscommon::calculate_temperature_from_virtual_temperature(team,T_virtual,qv,T_mid_from_virt);
+      physicscommon::calculate_virtual_temperature(team,temperature_packed,qv_packed,T_virtual_packed);
+      physicscommon::calculate_temperature_from_virtual_temperature(team,T_virtual_packed,qv_packed,T_mid_from_virt_packed);
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team,num_levs), [&] (const int k)
+      {
+        // T and TH conversion TEST
+        EKAT_KERNEL_REQUIRE(theta(k)==physicscommon::calculate_theta_from_T(temperature(k),pressure(k)));
```

Comment:
Since you are already doing it for other views, I would avoid doing checks on device. Instead, deep copy to host, and then use the REQUIRE macro on the host mirror.